Geometry2d/Point.hpp

```diff
@@ -157,6 +163,24 @@ class Point {
         return x() != other.x() || y() != other.y();
     }
 
+    double operator[](int i) {
```

Comment:
To fit more use cases, returning a reference would make it act more normally.
For example point[0] = 1 would do absolutely nothing right now and would probs confuse people.